### ISSUE TITLE
after finishing of a program throws this

### ISSUE THREAD

**Issue by fate8383** [thoukydides/homebridge-homeconnect#144]
### Description of Issue

[4/23/2023, 6:37:59 PM] [HomeConnect] Unexpected structure of Home Connect API event:
[4/23/2023, 6:37:59 PM] [HomeConnect] GET /api/homeappliances/events
[4/23/2023, 6:37:59 PM] [HomeConnect] └─ event is none of EventKeepAlive, EventApplianceConnection, EventApplianceData
[4/23/2023, 6:37:59 PM] [HomeConnect]     └─ event is not a EventApplianceData
[4/23/2023, 6:37:59 PM] [HomeConnect]         └─ event.data.items[0] is not a EventData
[4/23/2023, 6:37:59 PM] [HomeConnect]             └─ event.data.items[0].value is not a Value
[4/23/2023, 6:37:59 PM] [HomeConnect]                 └─ event.data.items[0].value is none of string, number, boolean
[4/23/2023, 6:37:59 PM] [HomeConnect] [API event stream] Home Connect API Validation Error: Structure validation failed (event is none of EventKeepAlive, EventApplianceConnection, EventApplianceData)
[4/23/2023, 6:37:59 PM] [HomeConnect] GET /api/homeappliances/events

### Expected Behaviour

should work without this API event error but not critical.

### Steps to Reproduce

run a simple program in the oven regardless how long it is, logs count down and at then it retrieves the device status as below but throws error at the end, however still functions as it for the next program or all other features. 

### Plugin Version

latest

### Environment

- OS: rpi4 debian latest stable
- Node: 18 latest stable
- Homebridge: latest stable

### Home Connect Appliance(s)

SIEMENS Oven HN678G4S6

### Diagnostic Checks

- [ ] Home Connect app is able to control the appliance with Wi-Fi disabled.
- [ ] Home Connect app shows three green lines in the Network section of the appliance Settings.
- [ ] Unofficial [Home Connect Server Status](https://homeconnect.thouky.co.uk) shows servers up and no recent errors.
- [ ] Searched for [previously reported issues](https://github.com/thoukydides/homebridge-homeconnect/issues?q=is%3Aissue) and no similar ones found.
- [ ] Checked the [Home Connect API errors](https://api-docs.home-connect.com/general?#api-errors) and [authorisation errors](https://api-docs.home-connect.com/authorization?#authorization-errors) documentation.

### Log File

```shell
[4/23/2023, 6:37:13 PM] [HomeConnect] [Oven] Program has 15 seconds remaining
[4/23/2023, 6:37:16 PM] [HomeConnect] [Oven] Program has 12 seconds remaining
[4/23/2023, 6:37:20 PM] [HomeConnect] [Oven] Program has 8 seconds remaining
[4/23/2023, 6:37:23 PM] [HomeConnect] [Oven] Program has 5 seconds remaining
[4/23/2023, 6:37:27 PM] [HomeConnect] [Oven] Program has 1 second remaining
[4/23/2023, 6:37:28 PM] [HomeConnect] [Oven] Inactive (Finished)
[4/23/2023, 6:37:28 PM] [HomeConnect] [Oven] No program running
[4/23/2023, 6:37:30 PM] [HomeConnect] [Oven] Manual mode (remote control activated, remote start allowed, being operated locally)
[4/23/2023, 6:37:30 PM] [HomeConnect] [Oven] Event Program Finished
[4/23/2023, 6:37:31 PM] [HomeConnect] [Oven] Door open
[4/23/2023, 6:37:46 PM] [HomeConnect] [Oven] Remote operation enabled (remote control activated, remote start allowed)
[4/23/2023, 6:37:59 PM] [HomeConnect] [Oven] Manual mode (remote control activated, remote start allowed, being operated locally)
[4/23/2023, 6:37:59 PM] [HomeConnect] [Oven] Event Program Finished off (ignored)
[4/23/2023, 6:37:59 PM] [HomeConnect] Unexpected structure of Home Connect API event:
[4/23/2023, 6:37:59 PM] [HomeConnect] GET /api/homeappliances/events
[4/23/2023, 6:37:59 PM] [HomeConnect] └─ event is none of EventKeepAlive, EventApplianceConnection, EventApplianceData
[4/23/2023, 6:37:59 PM] [HomeConnect]     └─ event is not a EventApplianceData
[4/23/2023, 6:37:59 PM] [HomeConnect]         └─ event.data.items[0] is not a EventData
[4/23/2023, 6:37:59 PM] [HomeConnect]             └─ event.data.items[0].value is not a Value
[4/23/2023, 6:37:59 PM] [HomeConnect]                 └─ event.data.items[0].value is none of string, number, boolean
[4/23/2023, 6:37:59 PM] [HomeConnect] [API event stream] Home Connect API Validation Error: Structure validation failed (event is none of EventKeepAlive, EventApplianceConnection, EventApplianceData)
[4/23/2023, 6:37:59 PM] [HomeConnect] GET /api/homeappliances/events
[4/23/2023, 6:37:59 PM] [HomeConnect] Starting events stream for all appliances
[4/23/2023, 6:37:59 PM] [HomeConnect] EVENT STREAM ERROR: AbortError: The operation was aborted
[4/23/2023, 6:37:59 PM] [HomeConnect] [Oven] Inactive (Inactive)
[4/23/2023, 6:37:59 PM] [HomeConnect] [Oven] Off
[4/23/2023, 6:37:59 PM] [HomeConnect] [Oven] Disconnected
[4/23/2023, 6:37:59 PM] [HomeConnect] [Oven] Off (disconnected)
[4/23/2023, 6:37:59 PM] [HomeConnect] [Oven] Inactive, attention required, in error state (disconnected)
[4/23/2023, 6:38:00 PM] [HomeConnect] [Oven] Door closed
[4/23/2023, 6:38:00 PM] [HomeConnect] [Oven] Manual mode (remote control activated, remote start allowed, being operated locally)
[4/23/2023, 6:38:00 PM] [HomeConnect] [Oven] Inactive, attention required, in error state (disconnected)
[4/23/2023, 6:38:01 PM] [HomeConnect] [Oven] Child lock disabled
[4/23/2023, 6:38:01 PM] [HomeConnect] [Oven] Alarm clock inactive
[4/23/2023, 6:38:01 PM] [HomeConnect] [Oven] Sabbath Mode off
[4/23/2023, 6:38:01 PM] [HomeConnect] [Oven] Off (disconnected)
[4/23/2023, 6:38:02 PM] [HomeConnect] [Oven] Connected
[4/23/2023, 6:38:02 PM] [HomeConnect] [Oven] Off
[4/23/2023, 6:38:02 PM] [HomeConnect] [Oven] Inactive (Inactive)
[4/23/2023, 6:38:13 PM] [HomeConnect] [Oven] Remote operation enabled (remote control activated, remote start allowed)
```

### Configuration

```json
{
            "clientid": "XXXX",
            "simulator": false,
            "language": {
                "api": "en-US"
            },
            "SIEMENS-HN678G4S6-XXX": {
                "addprograms": "none",
                "programs": []
            },
            "platform": "HomeConnect"
        },
```

**Comment by thoukydides** [comment]
Thanks for the report.

Reviewing the logs for my own instance I see the same issue. It is caused because the event from the Home Connect API includes:
```
"items": [
    {
        "handling": "none",
        "key": "BSH.Common.Root.ActiveProgram",
        "level": "hint",
        "timestamp": 1682007343,
        "uri": "/api/homeappliances/SIEMENS-SN678D06TG-68A40E096454/programs/active",
        "value": null
    },
```
where the `value` field is set to `null` rather than the name of a program.

In version v0.29.0 of this plugin I introduced checking of the API responses against the expected formats. This is one of the cases where the response does not match the documentation or OpenAPI/Swagger YAML schema... but had been handled appropriately by earlier versions of this plugin.

I should just need to add `null` as a valid `value`...

**Comment by thoukydides** [comment]
Should be fixed in v0.29.1.

**Comment by fate8383** [comment]
bunch of those errors are popping up today, not sure how to structure it what triggers them but I wanted to share here if it makes some sense to you. Thanks

[4/26/2023, 9:58:57 PM] [HomeConnect] Unexpected structure of Home Connect API event:
[4/26/2023, 9:58:57 PM] [HomeConnect] GET /api/homeappliances/events
[4/26/2023, 9:58:57 PM] [HomeConnect] └─ event is none of EventKeepAlive, EventApplianceConnection, EventApplianceData
[4/26/2023, 9:58:57 PM] [HomeConnect]     └─ event is not a EventApplianceData
[4/26/2023, 9:58:57 PM] [HomeConnect]         ├─ event.event is none of "STATUS", "EVENT", "NOTIFY"
[4/26/2023, 9:58:57 PM] [HomeConnect]         └─ event.data.items is missing
[4/26/2023, 9:58:57 PM] [HomeConnect] [API event stream] Home Connect API Validation Error: Structure validation failed (event is none of EventKeepAlive, EventApplianceConnection, EventApplianceData)
[4/26/2023, 9:58:57 PM] [HomeConnect] GET /api/homeappliances/events
[4/26/2023, 9:58:57 PM] [HomeConnect] Starting events stream for all appliances
[4/26/2023, 9:58:57 PM] [HomeConnect] EVENT STREAM ERROR: AbortError: The operation was aborted
[4/26/2023, 9:58:57 PM] [HomeConnect] [Oven] Disconnected
[4/26/2023, 9:58:57 PM] [HomeConnect] [Oven] Off (disconnected)
[4/26/2023, 9:58:57 PM] [HomeConnect] [Oven] Inactive, attention required, in error state (disconnected)
[4/26/2023, 9:59:57 PM] [HomeConnect] Unexpected structure of Home Connect API event:
[4/26/2023, 9:59:57 PM] [HomeConnect] GET /api/homeappliances/events
[4/26/2023, 9:59:57 PM] [HomeConnect] └─ event is none of EventKeepAlive, EventApplianceConnection, EventApplianceData
[4/26/2023, 9:59:57 PM] [HomeConnect]     └─ event is not a EventApplianceData
[4/26/2023, 9:59:57 PM] [HomeConnect]         ├─ event.event is none of "STATUS", "EVENT", "NOTIFY"
[4/26/2023, 9:59:57 PM] [HomeConnect]         └─ event.data.items is missing
[4/26/2023, 9:59:57 PM] [HomeConnect] [API event stream] Home Connect API Validation Error: Structure validation failed (event is none of EventKeepAlive, EventApplianceConnection, EventApplianceData)
[4/26/2023, 9:59:57 PM] [HomeConnect] GET /api/homeappliances/events
[4/26/2023, 9:59:57 PM] [HomeConnect] Starting events stream for all appliances
[4/26/2023, 9:59:57 PM] [HomeConnect] EVENT STREAM ERROR: AbortError: The operation was aborted
[4/26/2023, 9:59:58 PM] [HomeConnect] [Oven] Door closed
[4/26/2023, 9:59:58 PM] [HomeConnect] [Oven] Remote operation enabled (remote control activated, remote start allowed)
[4/26/2023, 9:59:58 PM] [HomeConnect] [Oven] Inactive, attention required, in error state (disconnected)
[4/26/2023, 9:59:58 PM] [HomeConnect] [Oven] Child lock disabled
[4/26/2023, 9:59:58 PM] [HomeConnect] [Oven] Alarm clock inactive
[4/26/2023, 9:59:58 PM] [HomeConnect] [Oven] Sabbath Mode off
[4/26/2023, 9:59:58 PM] [HomeConnect] [Oven] Off (disconnected)
[4/26/2023, 9:59:58 PM] [HomeConnect] [Oven] Connected
[4/26/2023, 9:59:58 PM] [HomeConnect] [Oven] Off
[4/26/2023, 9:59:58 PM] [HomeConnect] [Oven] Inactive (Inactive)
[4/26/2023, 10:01:18 PM] [HomeConnect] Unexpected structure of Home Connect API event:
[4/26/2023, 10:01:18 PM] [HomeConnect] GET /api/homeappliances/events
[4/26/2023, 10:01:19 PM] [HomeConnect] └─ event is none of EventKeepAlive, EventApplianceConnection, EventApplianceData
[4/26/2023, 10:01:19 PM] [HomeConnect]     └─ event is not a EventApplianceData
[4/26/2023, 10:01:19 PM] [HomeConnect]         ├─ event.event is none of "STATUS", "EVENT", "NOTIFY"
[4/26/2023, 10:01:19 PM] [HomeConnect]         └─ event.data.items is missing
[4/26/2023, 10:01:19 PM] [HomeConnect] [API event stream] Home Connect API Validation Error: Structure validation failed (event is none of EventKeepAlive, EventApplianceConnection, EventApplianceData)
[4/26/2023, 10:01:19 PM] [HomeConnect] GET /api/homeappliances/events
[4/26/2023, 10:01:19 PM] [HomeConnect] Starting events stream for all appliances
[4/26/2023, 10:01:19 PM] [HomeConnect] EVENT STREAM ERROR: AbortError: The operation was aborted
[4/26/2023, 10:01:19 PM] [HomeConnect] [Oven] Disconnected
[4/26/2023, 10:01:19 PM] [HomeConnect] [Oven] Off (disconnected)
[4/26/2023, 10:01:19 PM] [HomeConnect] [Oven] Inactive, attention required, in error state (disconnected)

**Comment by thoukydides** [comment]
It looks like this is probably due to the `CONNECTED` and `DISCONNECTED` events not providing data in the format defined by the OpenAPI/Swagger YAML schema (and the format is not actually described anywhere else in the API documentation). For both this and the end of program issue, the real appliances also return the information in a different format to the appliance simulators...

**Comment by thoukydides** [comment]
Should be fixed in v0.29.2.
